src/app/clusters/energy-evse-server/energy-evse-server.cpp

```diff
@@ -0,0 +1,438 @@
+        return aEncoder.Encode(mDelegate.GetSessionDuration());
+    case SessionEnergyCharged::Id:
+        return aEncoder.Encode(mDelegate.GetSessionEnergyCharged());
+    case SessionEnergyDischarged::Id:
+        return aEncoder.Encode(mDelegate.GetSessionEnergyDischarged());
+    }
+    /* Allow all other unhandled attributes to fall through to Ember */
+    return CHIP_NO_ERROR;
+}
+
+CHIP_ERROR Instance::Write(const ConcreteDataAttributePath & aPath, AttributeValueDecoder & aDecoder)
+{
+    switch (aPath.mAttributeId)
+    {
+    case UserMaximumChargeCurrent::Id: {
+        // Optional Attribute
+        if (!SupportsOptAttr(OptionalAttributes::kSupportsUserMaximumChargingCurrent))
+        {
+            return CHIP_IM_GLOBAL_STATUS(UnsupportedAttribute);
+        }
+
+        int64_t newValue;
+        ReturnErrorOnFailure(aDecoder.Decode(newValue));
+        ReturnErrorOnFailure(mDelegate.SetUserMaximumChargeCurrent(newValue));
+        return CHIP_NO_ERROR;
+    }
+    case RandomizationDelayWindow::Id: {
+        // Optional Attribute
+        if (!SupportsOptAttr(OptionalAttributes::kSupportsRandomizationWindow))
+        {
+            return CHIP_IM_GLOBAL_STATUS(UnsupportedAttribute);
+        }
+        uint32_t newValue;
+        ReturnErrorOnFailure(aDecoder.Decode(newValue));
+        ReturnErrorOnFailure(mDelegate.SetRandomizationDelayWindow(newValue));
+        return CHIP_NO_ERROR;
+    }
+    case ApproximateEVEfficiency::Id: {
+        // Optional Attribute if ChargingPreferences is supported
+        if ((!HasFeature(Feature::kChargingPreferences)) ||
+            (!SupportsOptAttr(OptionalAttributes::kSupportsApproximateEvEfficiency)))
+        {
+            return CHIP_IM_GLOBAL_STATUS(UnsupportedAttribute);
+        }
+        uint16_t newValue;
+        ReturnErrorOnFailure(aDecoder.Decode(newValue));
+        ReturnErrorOnFailure(mDelegate.SetApproximateEVEfficiency(newValue));
+        return CHIP_NO_ERROR;
+    }
+
+    default:
+        // Unknown attribute; return error.  None of the other attributes for
+        // this cluster are writable, so should not be ending up in this code to
+        // start with.
+        return CHIP_IM_GLOBAL_STATUS(UnsupportedAttribute);
+    }
+}
+
+// CommandHandlerInterface
+
+CHIP_ERROR Instance::EnumerateAcceptedCommands(const ConcreteClusterPath & cluster, CommandIdCallback callback, void * context)
+{
+    using namespace Commands;
+
+    for (auto && id : {
+             Disable::Id,
+             EnableCharging::Id,
+         })
+    {
+        if (callback(id, context) == Loop::Break)
+        {
+            return CHIP_NO_ERROR;
+        }
+    }
+
+    if (HasFeature(Feature::kV2x))
+    {
+        callback(EnableDischarging::Id, context);
+    }
+    if (HasFeature(Feature::kChargingPreferences))
+    {
+        callback(SetTargets::Id, context);
+        callback(GetTargets::Id, context);
+        callback(ClearTargets::Id, context);
```

Comment:
If `callback` returns `Loop::Break` there must be no more calls to it....

Reply:
ah - ok - will put in multiple loops.

Reply:
ok - but there could be a scenario where not all supported command get registered?

Why do all the examples return CHIP_NO_ERROR in this situation?
eg network-commissioning.cpp 

Is there a bug there?

Reply:
Ensured it exits now if ! Loop::Continue based on using network-commissioning example:
        VerifyOrExit(callback(cmd, context) == Loop::Continue, /**/);